Assets/MixedRealityToolkit/Providers/UnityInput/UnityTouchController.cs

```diff
@@ -79,22 +79,35 @@ public override void SetupDefaultInteractions(Handedness controllerHandedness)
             }
         }
 
+        private bool isNewController = false;
+
         /// <summary>
         /// Start the touch.
         /// </summary>
         public void StartTouch()
         {
-            InputSystem?.RaisePointerDown(InputSource.Pointers[0], Interactions[2].MixedRealityInputAction);
-            isTouched = true;
-            InputSystem?.RaiseGestureStarted(this, holdingAction);
-            isHolding = true;
+            // Indicate that this is a new controller.
+            isNewController = true;
         }
 
         /// <summary>
         /// Update the touch data.
         /// </summary>
         public void Update()
         {
+            if (InputSystem == null) { return; }
+
+            if (isNewController)
+            {
+                isNewController = false;
+
+                InputSystem.RaiseOnInputDown(InputSource, Handedness.None, Interactions[2].MixedRealityInputAction);
+                InputSystem.RaisePointerDown(InputSource.Pointers[0], Interactions[2].MixedRealityInputAction);
```

Comment:
We should definitely move these into the pointer at some point. Don't need to block this PR on it though, since it's existing behavior.

Reply:
100%